src/condor_tools/htcondor_cli/eventlog.py

```diff
@@ -0,0 +1,117 @@
+import os
+import re
+
+from pathlib import Path
+
+import htcondor
+
+from htcondor_cli.noun import Noun
+from htcondor_cli.verb import Verb
+from datetime import datetime
+from datetime import timedelta
+
+class Read(Verb):
+    """
+    Reads an eventlog
+    """
+
+    options = {
+        "log_file": {
+            "args": ("log_file",),
+            "help": "Log file",
+        }
+    }
+
+    def __init__(self, logger, log_file, **options):
+        # make sure the specified log file exists and is readable
+        log_file = Path(log_file)
+        if not log_file.exists():
+            raise FileNotFoundError(f"Could not find file: {str(log_file)}")
+        if os.access(log_file, os.R_OK) is False:
+            raise PermissionError(f"Could not access file: {str(log_file)}")
+        
+        # dictionary to store summary of events for each job
+        event_summaries = {}
+
+        with htcondor.JobEventLog(str(log_file)) as event_log:
+            for event in event_log.events(0):
+                job_id = str(event.cluster) + "." + str(event.proc)
+                event_summary = event_summaries.get(job_id, {})
+                if event.type == htcondor.JobEventType.SUBMIT:
+                    ip_address = re.search(r"\d+\.\d+\.\d+\.\d+", event.get("SubmitHost")).group()
+                    event_summaries[job_id] = {"Host": ip_address, "Evictions": 0, "Executed": False}
+                elif event.type == htcondor.JobEventType.JOB_ABORTED and event_summary.get("Executed", False) is False:
+                    # remove job from dictionary if it was aborted before being executed
+                    event_summaries.pop(job_id)
+                elif event.type == htcondor.JobEventType.EXECUTE:
+                    # get hostname if possible
+                    if event.get("SlotName") is not None:
+                        hostname = event.get("SlotName")
+                        event_summary["Host"] = hostname
```

Comment:
You know that the key `"SlotName"` has a value, so you could save a line of code:
```suggestion
                        event_summary["Host"] = event["SlotName"]
```